gnosis/gp-swap-ui
# Wrap eth v2

# Summary

Continues the work on and supersedes https://github.com/gnosis/gp-swap-ui/pull/72

Implemented wrapping ETH when selling.

When `from` is set to `ETH`, 2 wallet pop ups should be displayed in sequence to the user:

1. Wrap necessary amount to complete the order of `ETH` into `WETH`. Onchain
2. Place order. Offchain

![screenshot_2021-01-08_13-33-56](https://user-images.githubusercontent.com/43217/104070207-341d4600-51bb-11eb-84d6-01e49a24ed43.png)

Works both with fixed `from` ("sell" order) and fixed `to` ("buy" order) amounts.

# To test

1. Fire the app
2. Select ETH on `from`, and anything else except WETH on `to`
3. Click `Swap` and confirm.

- [ ] The first pop up should be the one to wrap ETH
- [ ] The second pop up should be to sign the offline order

# Caveats

- :warning:  Does NOT unwrap WETH when receiving ETH (`ETH` set on `to` field)
- If you cancel the first request, you'll still be prompted for, and can sign the offline order
